corehq/apps/linked_domain/updates.py

```diff
@@ -299,15 +307,47 @@ def update_user_roles(domain_link):
             ]
         downstream_role.set_assignable_by(assignable_by)
 
+    if failed_updates:
+        conflicting_role_names = ', '.join(['"{}"'.format(role['name']) for role in failed_updates])
+        error_msg = _('Failed to sync the following roles due to a conflict: {}.'
+            ' Please remove or rename these roles before syncing again').format(conflicting_role_names)
+        raise UnsupportedActionError(error_msg)
+
+
+def _get_matching_downstream_role(upstream_role_json, downstream_roles):
+    matching_role = None
+    conflicting_role = None
+
+    for downstream_role in downstream_roles:
+        if upstream_role_json['_id'] == downstream_role.upstream_id:
+            matching_role = downstream_role
+        elif upstream_role_json['name'] == downstream_role.name:
+            if _is_default_built_in_role(upstream_role_json, downstream_role):
+                matching_role = downstream_role
+            else:
+                conflicting_role = downstream_role
+
+    return (matching_role, conflicting_role)
```

Comment:
nit: just curious why you chose this instead of a tuple that returned `(role, is_conflict)`

Reply:
Good catch. I believe I initially implemented it as you're suggesting, then ended up needing the specific name of the role that conflicted. I don't think that's necessary anymore, so I'm going to adopt what you're suggesting here